cypress/elements/layout.js

```diff
@@ -0,0 +1,119 @@
+const youCategorySelectOptionEl = 'yoy-layout-rows-select-option'
+const chipEl = 'layout-chip'
+const chipMenuButtonEl = 'layout-chip-menu-button'
+const chipMenuRemoveOptionEl = 'layout-chip-menu-dimension-menu-item-remove'
+const chipMenuActionOptionEl = 'layout-chip-menu-dimension-menu-item-action'
+const chipMenuSubMenuOptionEl =
+    'layout-chip-menu-dimension-menu-item-DIMENSIONID-menu'
+const chipSelectedBackgroundColor = 'rgb(178, 223, 219)'
+
+const getAxisEl = axisId => `${axisId}-axis`
+const getDimensionChipEl = dimensionId => `${chipEl}-${dimensionId}`
+
+export const selectYoyCategoryOption = option => {
+    cy.getBySel(yoyCategorySelectEl).click()
+    cy.getBySelLike(youCategorySelectOptionEl)
+        .contains(option)
+        .click()
+}
+
+export const expectAxisToHaveDimension = (axisId, dimensionId) => {
+    if (axisId && dimensionId) {
+        cy.getBySel(getAxisEl(axisId))
+            .findBySel(getDimensionChipEl(dimensionId))
+            .should('have.length', 1)
+            .and('be.visible')
+    } else {
+        throw new Error('axisId and dimensionId not provided')
+    }
+}
+
+export const expectAxisToNotHaveDimension = (axisId, dimensionId) => {
+    if (axisId && dimensionId) {
+        cy.getBySel(getAxisEl(axisId))
+            .findBySel(getDimensionChipEl(dimensionId))
+            .should('have.length', 0)
+            .and('not.be.visible')
+    } else {
+        throw new Error('axisId and dimensionId not provided')
+    }
+}
+
+export const expectLayoutToHaveDimension = dimensionId =>
+    cy
+        .getBySel(getDimensionChipEl(dimensionId))
+        .should('have.length', 1)
+        .and('be.visible')
+
+export const expectLayoutToNotHaveDimension = dimensionId =>
+    cy
+        .getBySel(getDimensionChipEl(dimensionId))
+        .should('have.length', 0)
+        .and('not.be.visible')
+
+export const expectDimensionToHaveItemAmount = (
+    dimensionId,
+    itemAmount = 0
+) => {
+    if (dimensionId) {
+        itemAmount
+            ? cy
+                  .getBySel(getDimensionChipEl(dimensionId))
+                  .contains(`${itemAmount} selected`)
+                  .should('have.length', 1)
+                  .and('be.visible')
+            : cy
+                  .getBySel(getDimensionChipEl(dimensionId))
+                  .contains(`selected`)
+                  .should('have.length', 0)
+    } else {
+        throw new Error('dimensionId not provided')
+    }
+}
+
+export const expectDimensionToNotHaveItems = dimensionId =>
+    expectDimensionToHaveItemAmount(dimensionId)
+
+export const openDimension = dimensionId =>
+    cy.getBySel(getDimensionChipEl(dimensionId)).click()
+
+export const openContextMenu = dimensionId =>
+    cy.getBySel(`${chipMenuButtonEl}-${dimensionId}`).click()
+
+export const clickContextMenuAdd = (dimensionId, axisId) =>
+    cy
+        .getBySel(`${chipMenuActionOptionEl}-${dimensionId}-to-${axisId}`)
+        .should('contain', 'Add to')
```

Comment:
Could this be `.contains('Add to').click()` instead? Since this is an action function, not an expectation function.  (same comment for other places in this file)

Reply:
This is actually an assertion to check that the option says `Add to` and not `Move to`, because the option label switches between "add" and "move" if the dimension is in the layout already or not. `contains` would also work and since its an action it might be better to not use an assertion here.